### PR TITLE
[vcpkg] Two bug fixes and some improvements in buildsystems/msbuild

### PR DESCRIPTION
**Pull request by FrankHeimes**
Applied changes as described in #13753
- Fixed: Empty fields in Vcpkg property sheet
- Fixed: Missing `-static` suffix in `VcpkgCurrentInstalledDir`
- Removed redundant property definitions
- Various small improvements

### CODE REVIEW

**Comment by FrankHeimes** [comment]
I have no clue why `scripts\testing\integrate-install\VcpkgTriplet.vcxproj` keeps failing to locate `zlib.h` when compiling. If I load and build the project in Visual Studio, it builds just fine. The properties `VcpkgRoot` and `VcpkgCurrentInstalledDir` have their correct values, so the compiler can find `zlib.h` in `$(VcpkgCurrentInstalledDir)include\zlib.h`.

**Comment by JackBoosY** [comment]
Please merge to master branch.

**Comment by FrankHeimes** [comment]
As I see it, I'm done with my changes and fixes and humbly awaiting your code review remarks.

File: scripts/buildsystems/msbuild/vcpkg.props [hunk 1]
```diff
@@ -1,41 +1,34 @@
...
+    <VcpkgAutoLink Condition="'$(VcpkgAutoLink)' == ''">true</VcpkgAutoLink>
+    <!-- Deactivate Autolinking if lld is used as a linker. (Until a better way to solve the problem is found!).
+    Tried to add /lib as a parameter to the linker call but was unable to find a way to pass it as the first parameter. -->
+    <VcpkgAutoLink Condition="'$(UseLldLink)' == 'true'">false</VcpkgAutoLink>
+
+    <!-- Manifest files -->
     <VcpkgEnableManifest Condition="'$(VcpkgEnableManifest)' == ''">false</VcpkgEnableManifest>
     <VcpkgManifestInstall Condition="'$(VcpkgManifestInstall)' == ''">true</VcpkgManifestInstall>
-    <VcpkgManifestRoot>$([MSbuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), vcpkg.json))</VcpkgManifestRoot>
+    <VcpkgManifestRoot Condition="'$(VcpkgManifestRoot)' != ''">$([MSbuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), vcpkg.json))"</VcpkgManifestRoot>
```
Comment (strega-nil) [inline scripts/buildsystems/msbuild/vcpkg.props:31]:
```suggestion
    <VcpkgManifestRoot Condition="'$(VcpkgManifestRoot)' != ''">$([MSbuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), vcpkg.json))</VcpkgManifestRoot>
```
stray `"`?

**Review by strega-nil — COMMENTED** [review]
This looks like the only thing to me, thanks @FrankHeimes :)

Could you also merge with master?

File: scripts/buildsystems/msbuild/vcpkg.props (hunk 1 above)
Comment (FrankHeimes) [reply scripts/buildsystems/msbuild/vcpkg.props]: Thanks for reviewing! Fixed.

**Comment by JackBoosY** [comment]
Please solve the file conflict,

Thanks.

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil** [comment]
This LGTM; however, I don't _really_ understand MSBuild, so I'm just going to try it out and see if it works.

**Comment by strega-nil** [comment]
Looks like it works, although I think there's one thing I'd like to change.

**Pull request merged**